tonic/src/transport/server/tls.rs

```diff
@@ -38,15 +38,29 @@ impl ServerTlsConfig {
         }
     }
 
-    /// Sets a certificate against which to validate client TLS certificates.
-    pub fn client_ca_root(self, cert: Certificate) -> Self {
+    /// Sets the [`ClientTlsAuth`] approach.
+    pub fn client_auth(self, client_auth: ClientTlsAuth) -> Self {
         ServerTlsConfig {
-            client_ca_root: Some(cert),
+            client_auth,
             ..self
         }
     }
 
     pub(crate) fn tls_acceptor(&self) -> Result<TlsAcceptor, crate::Error> {
-        TlsAcceptor::new(self.identity.clone().unwrap(), self.client_ca_root.clone())
+        TlsAcceptor::new(self.identity.clone().unwrap(), self.client_auth.clone())
     }
 }
+
+/// Client TLS authentication options.
+#[cfg(feature = "tls")]
+#[cfg_attr(docsrs, doc(cfg(feature = "tls")))]
+#[derive(Clone, Debug, Default)]
+pub enum ClientTlsAuth {
```

Comment:
This should be a private enum so that we don't add additional api surface area.

Reply:
This is not used anymore.